### PR TITLE
feat(autoscaling): support for throughput on GP3 volumes

### PR DESCRIPTION
**Pull request by csumpter**
Adds support for the `throughput` property on GP3 volumes in both autoscaling and EC2 packages since their volume implementations are separate per the comment https://github.com/aws/aws-cdk/blob/master/packages/@aws-cdk/aws-autoscaling/lib/volume.ts#L1.

Change includes unit test coverage, integration test coverage on the autoscaling changes, and validation of the inputq to throughput. I was on the fence about whether to include the validation for synth time checking since as far as I can tell validation is not performed consistently across the CDK at synth time. Happy to modify that behavior pending reviewer feedback.

It was not obvious to me at first pass where similar integration test coverage that was added to the autoscaling package could be added to the EC2 package, so if the reviewer would like to see integration tests for GP3 volumes on in the EC2 package, would you mind providing a hint as to where that test might fit in the existing test paradigm?

Happy to break this change into two PRs -- one for EC2 and one for autoscaling.

closes: https://github.com/aws/aws-cdk/issues/16213

----

### All Submissions:

* [x] Have you followed the guidelines in our [Contributing guide?](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md)

### Adding new Unconventional Dependencies:

* [ ] This PR adds new unconventional dependencies following the process described [here](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md/#adding-new-unconventional-dependencies)

### New Features

* [x] Have you added the new feature to an [integration test](https://github.com/aws/aws-cdk/blob/main/INTEGRATION_TESTS.md)?
	* [x] Did you use `yarn integ` to deploy the infrastructure and generate the snapshot (i.e. `yarn integ` without `--dry-run`)?

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk/aws-autoscaling/lib/volume.ts [hunk 1]
```diff
@@ -66,6 +66,13 @@ export interface EbsDeviceOptionsBase {
    * @default {@link EbsDeviceVolumeType.GP2}
    */
   readonly volumeType?: EbsDeviceVolumeType;
+
+  /**
+   * The throughput that the volume supports, in MiB/s
```
Comment (corymhall) [inline packages/@aws-cdk/aws-autoscaling/lib/volume.ts:71]: Can you add some docs here around min/max value? Also I think if you increase it above 750 you also
have to increase the iops. Maybe we should automatically set the iops based on throughput _if_ a
specific iops is not provided? Also add something that says this only applies to GP3

**Review by corymhall — DISMISSED** [review]
Just a couple of comments

File: packages/@aws-cdk/aws-autoscaling/lib/volume.ts (hunk 1 above)
Comment (csumpter) [reply packages/@aws-cdk/aws-autoscaling/lib/volume.ts:71]: Thank you for the comments @corymhall -- just seeing this now.

> Can you add some docs here around min/max value?

Happy to!

> Maybe we should automatically set the iops based on throughput if a
specific iops is not provided?

Do you know if cloudformation / the backing API will return a specific error if you turn it above 750 w/o an iops input or will the backing service just take a sane default? I can also poke at the system behavior and once I figure it out code it into the tests.

I will try to find the docs on this.

> Also add something that says this only applies to GP3

Sure thing!

Comment (csumpter) [reply packages/@aws-cdk/aws-autoscaling/lib/volume.ts:71]: @corymhall - the requested changes have been added to the PR, please take a look when you get a chance.

**Comment by csumpter** [comment]
Hello aws-cdk team! This PR is ready for review at your leisure. Please let me know if there are any additional updates you would like to see. Thank you!

**Review by mrgrain — DISMISSED** [review]
Looks all good now! 🚀 

Thanks for the contribution @csumpter

**Review by mrgrain — APPROVED** [review]

**Pull request merged**